internal/app/proxy_stage_test.go

```diff
@@ -356,126 +351,92 @@ func (s *ProxyStage) multiple_requests_are_sent(requestsToSend int32) {
 			for i := int32(0); i < requestsToSend; i++ {
 				u := fmt.Sprintf("http://localhost:%s/users", proxyURL.Port())
 				req, err := http.NewRequest("POST", u, strings.NewReader(`{"name":"test"}`))
-				if err != nil {
-					s.t.Error(err)
-					s.t.Fail()
-				}
+				s.assert.NoError(err)
 
 				req.Header.Set("Content-Type", "application/json")
 				atomic.AddInt32(&s.requestsSent, 1)
-				if _, err = http.DefaultClient.Do(req); err != nil {
-					s.t.Error(err)
-					s.t.Fail()
-				}
+				_, err = http.DefaultClient.Do(req)
+				s.assert.NoError(err)
+				// s.t.Fatal("oops") // This exits the inner func instead of TestFoo.
+				// fatal(s)
 			}
 		}()
 
-		if err := s.proxy.WaitForInteraction(s.pactName, int(requestsToSend)); err != nil {
-			s.t.Error(err)
-			s.t.Fail()
-		}
+		go func() {
```

Comment:
:confused: not sure what is going on here?

Reply:
removed